src/apportioned-emissions/quarterly/quarter-unit-data.repository.ts

```diff
@@ -68,4 +68,89 @@ export class QuarterUnitDataRepository extends Repository<QuarterUnitDataView> {
 
     return query;
   }
+
+  async getEmissionsFacilityAggregation(
+    req: Request,
+    params: PaginatedQuarterlyApportionedEmissionsParamsDTO,
+  ): Promise<QuarterUnitDataView[]> {
+    let totalCount: number;
+    let results: QuarterUnitDataView[];
+    const { page, perPage } = params;
+
+    const selectColumns = [
+      'aud.stateCode',
+      'aud.facilityName',
+      'aud.facilityId',
+      'aud.year',
+      'aud.quarter'
+    ];
+    const orderByColumns = ['aud.facilityId', 'aud.year', 'aud.quarter'];
+
+    const query = this.buildAggregationQuery(
+      params,
+      selectColumns,
+      orderByColumns,
+    );
+
+    results = await query.getRawMany();
+    if (page && perPage) {
+      const countQuery = this.buildAggregationQuery(
+        params,
+        selectColumns,
+        orderByColumns,
+        true,
+      );
+      totalCount = (await countQuery.getRawOne()).count;
+      ResponseHeaders.setPagination(req, page, perPage, totalCount);
+    }
+    return results;
+  }
+
+  private buildAggregationQuery(
+    params,
+    selectColumns: string[],
+    orderByColumns: string[],
+    countQuery: boolean = false,
+  ): SelectQueryBuilder<QuarterUnitDataView> {
+    let query = null;
+
+    if (countQuery) {
+      query = this.createQueryBuilder('aud').select('COUNT(*) OVER() as count');
```

Comment:
Since we are pulling from quarter unit data view the alias should reflect taht and be 'qud'. The alias below need to be changed as well

Reply:
That makes sense. I've implemented the update!